components/dropdown/test/dropdown-content.visual-diff.js

```diff
@@ -0,0 +1,56 @@
+const helper = require('./dropdown-helper.js');
+const puppeteer = require('puppeteer');
+const VisualDiff = require('@brightspace-ui/visual-diff');
+
+describe('d2l-dropdown-content', () => {
+
+	const visualDiff = new VisualDiff('dropdown-content', __dirname);
+
+	let browser, page;
+
+	before(async() => {
+		browser = await puppeteer.launch();
+		page = await browser.newPage();
+		await page.setViewport({width: 800, height: 400, deviceScaleFactor: 2});
+		await page.goto(`${visualDiff.getBaseUrl()}/components/dropdown/test/dropdown-content.visual-diff.html`, {waitUntil: ['networkidle0', 'load']});
+		await page.bringToFront();
+	});
+
+	after(() => browser.close());
+
+	[
+		'default-width',
+		'min-width',
+		'default-max-width',
+		'max-width',
+		'scroll-bottom-shadow',
+		'top-middle',
```

Comment:
I think there may be a permutation of positioning that art not covered.  For positioning, I think we should have:

* centered with target (dropdown content is wider than opener)
* centered with target (dropdown content is narrower than opener)
* slide to right (not enough space to center/left edge of page)
* slide to left (not enough space to center/right edge of page)
* ^ all the above with RTL
* open upwards (when not enough space below, but there is above)

I think I would name them according to the behavior.  So for example, I think `top-right` should be named something like `slide-left`.

I'm not quite as familiar with the boundary usage, but I _think_ they are covered here.

Reply:
Just taking a second look at this.  Good job on the images/fixture setup by the way!  

I think the only scenario(s) not covered are the "dropdown content is narrower than opener", and some RTL. 

While the RTL cases are visually opposite of the LTR, the calculation for the RTL is done in a different code path since the property assignment is different, and calc for slide lift/right is flipped, so for complete position coverage we would want RTL captures for those scenarios.